### PR TITLE
[Silabs]Provision storage sourceset

### PR DESCRIPTION
**Pull request by jmartinez-silabs**
Add gn arg to select what provision/factory data storage implementation to use.

Create a source_set that deals with matching the correct storage implementation source and matching provision lib for efr32 or si917 platform

### CODE REVIEW

File: examples/platform/silabs/provision/BUILD.gn [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import("//build_overrides/efr32_sdk.gni")
+import("${efr32_sdk_build_root}/silabs_board.gni")
+
+if (wifi_soc) {
+  import("${efr32_sdk_build_root}/SiWx917_sdk.gni")
+} else {
+  import("${efr32_sdk_build_root}/efr32_sdk.gni")
+}
+
+source_set("provision-storage") {
```
Comment (mkardous-silabs) [inline examples/platform/silabs/provision/BUILD.gn:25]: This ends being provision:provision which seems a bit redundant. Might be worth while to remove the `provision-` bits.

File: third_party/silabs/efr32_sdk.gni [hunk 2]
```diff
@@ -82,6 +82,9 @@ declare_args() {
   chip_enable_multi_ota_requestor = false
   chip_enable_multi_ota_encryption = false
   chip_enable_ota_custom_tlv_testing = false
+
+  # Factory Provision storage
+  use_flash_storage = wifi_soc
```
Comment (mkardous-silabs) [inline third_party/silabs/efr32_sdk.gni:87]:
```suggestion
  sl_use_flash_storage = wifi_soc
```

File: third_party/silabs/efr32_sdk.gni [hunk 3]
```diff
@@ -469,6 +472,12 @@ template("efr32_sdk") {
       defines += [ "SL_USE_COAP_CONFIG=0" ]
     }
 
+    if (use_flash_storage) {
+      defines += [ "SL_PROVISION_FLASH=1" ]
+    } else {
+      defines += [ "SL_PROVISION_FLASH=0" ]
```
Comment (mkardous-silabs) [inline third_party/silabs/efr32_sdk.gni:478]: Can this be renamed to `SL_MATTER_PROVISION_FLASH`?

File: src/platform/silabs/efr32/efr32-chip-mbedtls-config.h [hunk 4]
```diff
@@ -87,6 +87,12 @@
 #define PSA_WANT_ALG_SOME_PAKE
 #endif // SL_USE_COAP_CONFIG
 
+#if SL_PROVISION_FLASH
+#define MBEDTLS_SHA256_C
+#define MBEDTLS_HMAC_DRBG_C
+#define MBEDTLS_ECDSA_DETERMINISTIC
+#endif
```
Comment (mkardous-silabs) [inline src/platform/silabs/efr32/efr32-chip-mbedtls-config.h:94]:
```suggestion
#endif // SL_PROVISION_FLASH
```

File: third_party/silabs/efr32_sdk.gni (hunk 2 above)
Comment (mkardous-silabs) [inline third_party/silabs/efr32_sdk.gni:87]: we should add `asserts` to make sure the flash/non flash makes sense with the crypto flavor.

Comment (jmartinez-silabs) [reply third_party/silabs/efr32_sdk.gni:87]: Technically they could/can be independent

**Review by mkardous-silabs — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Pull request merged**